### PR TITLE
CORS-3480: aws/sdk: remove sdk provisioner

### PR DESCRIPTION
**Pull request by r4f4**
It's been replaced by capi/capa as the terraform-free alternative.

Depends on https://github.com/openshift/installer/pull/8343
Depends on https://github.com/openshift/installer/pull/8357

### CODE REVIEW

**Comment by r4f4** [comment]
Update: rebased to fix merge conflicts.

File: pkg/infrastructure/platform/platform.go [hunk 1]
```diff
@@ -51,12 +50,9 @@ import (
 func ProviderForPlatform(platform string, fg featuregates.FeatureGate) (infrastructure.Provider, error) {
 	switch platform {
 	case awstypes.Name:
-		if types.ClusterAPIFeatureGateEnabled(platform, fg) {
+		if types.ClusterAPIFeatureGateEnabled(platform, fg) || fg.Enabled(features.FeatureGateInstallAlternateInfrastructureAWS) {
```
Comment (barbacbd) [inline pkg/infrastructure/platform/platform.go:53]: Can `features.FeatureGateInstallAlternateInfrastructureAWS` be removed now from the installer ?

Comment (r4f4) [reply pkg/infrastructure/platform/platform.go:53]: Yes. We can do that one before this PR, but it'll involve changes to `openshift/api` to remove the gate.

Comment (r4f4) [reply pkg/infrastructure/platform/platform.go:53]: Actually, if we just ignore the gate in the Installer, then later on we can remove it from `openshift/api` without having to bump the api version in the installer.

Comment (patrickdillon) [reply pkg/infrastructure/platform/platform.go:53]: I'm ok either way, but lean toward removing the `FeatureGateInstallAlternateInfrastructureAWS` feature gate.

**Comment by patrickdillon** [comment]
/approve

This LGTM. Do you want to keep `FeatureGateInstallAlternateInfrastructureAWS` (as the PR stands now)? Or should we remove it in this PR, so we can ultimately remove from the API?

**Comment by r4f4** [comment]
> This LGTM. Do you want to keep `FeatureGateInstallAlternateInfrastructureAWS` (as the PR stands now)? Or should we remove it in this PR, so we can ultimately remove from the API?

I'll update the PR to remove it. That will make removing it from openshift/api much easier.

**Review by barbacbd — COMMENTED** [review]
/lgtm

**Comment by patrickdillon** [comment]
let's try that again!

/label acknowledge-critical-fixes-only

**Pull request merged**